lib/IDE/TypeReconstruction.cpp

```diff
@@ -792,11 +792,11 @@ static void VisitNodeGenericTypealias(ASTContext *ast,
   Type parentType;
   if (auto nominal = genericTypeAlias->getDeclContext()
                          ->getAsNominalTypeOrNominalTypeExtensionContext()) {
-    parentType = nominal->getDeclaredInterfaceType();
+    parentType = nominal->getDeclaredInterfaceType().subst(subMap);
   }
-  NameAliasType *NAT =
-      NameAliasType::get(genericTypeAlias, parentType, subMap,
-                         genericTypeAlias->getUnderlyingTypeLoc().getType());
+  NameAliasType *NAT = NameAliasType::get(
+      genericTypeAlias, parentType, subMap,
```

Comment:
The parent type is still wrong, it should have the same substitution map applied